index.d.ts

```diff
@@ -0,0 +1,82 @@
+import { Action, StoreEnhancer, Store } from "redux";
+
+export interface StoreCreator {
+  <S, T>(
+    reducer: EnhancedReducer<S>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ): Store<S>;
+}
+
+export interface EnhancedReducer<S> {
+  <A extends Action>(state: S, action: A): S | [S, CmdType<A, Function>];
+}
+
+declare interface LiftedEnhancedReducer<S> {
```

Comment:
@bdwain okay - I got rid of the second generic argument to EnhancedReducer and then created LiftedEnhancedReducer to reflect cases like combineReducers where the return type will never just be `S` -- might be a more concise way of expressing this, but for a first pass this should work

Reply:
good idea!